### ISSUE TITLE
How sync files as like the dropbox app?

### ISSUE THREAD

**Issue by mamcx** [dropbox/SwiftyDropbox#133]
I need to provide a sync facility for some files. I wish to let dropbox do the work.

I integrate the api, but the end-result behave as a normal upload operation. If I have 5 files that need to sync, the api upload all of them again, instead of just sync the changes.

I wish to have the same behaviour as with the dropbox client. Is this possible with this API?

If not, then how I can send files to be openned in the native iOS client?

**Comment by greg-db** [comment]
The Dropbox API offers an interface for uploading and downloading files on demand, as you noted. It doesn't provide any sort of pre-built sync solution, but I'll be sure to pass this along as a feature request. That being the case, you'll need to implement whatever syncing logic you need for your app in your app's code using the Dropbox API.

For example, your app should upload files only when you have local changes to upload. (The API doesn't offer the capability to upload only the changed portions of an individual file though, as opposed to the entire file. Is that what you're looking for?)

There also isn't an interface for deep-linking or otherwise using the official iOS app to display files, but we'll consider that a feature request as well.

**Comment by mamcx** [comment]
I fear that. Is possible to offer the "Open in Dropbox" from my app?

**Comment by greg-db** [comment]
No, that would be a sort of "deep-linking" feature I was referring to, which the API doesn't offer. Apologies I don't have a better answer for you!

**Comment by bhhales** [comment]
Along similar lines, are there plans to add push notification support, so our apps can be notified when there are changes?

**Comment by greg-db** [comment]
No plans for that that I'm aware of, but I'll send this along as a feature request.

The closest we have for client-side apps is to use [`listFolderLongpoll`](https://dropbox.github.io/SwiftyDropbox/api-docs/latest/Classes/FilesRoutes.html#/s:FC13SwiftyDropbox11FilesRoutes18listFolderLongpollFT6cursorSS7timeoutVs6UInt64_GCS_10RpcRequestCCS_5Files34ListFolderLongpollResultSerializerCS3_33ListFolderLongpollErrorSerializer_) while the app is open. Or, if you have your own server, you can use [webhooks](https://www.dropbox.com/developers/reference/webhooks) and trigger push notifications from your server.
